### PR TITLE
feat: LT-1021 Licensed Quantity & Licence page

### PR DESCRIPTION
**Pull request by BenAndrew-Kainos**

### CODE REVIEW

File: core/builtins/custom_tags.py [hunk 1]
```diff
@@ -60,6 +62,29 @@ def str_date(value):
     )
 
 
+@register.filter
+@stringfilter
+def str_date_only(value):
+    date_str = do_timezone(datetime.datetime.strptime(value, DATE_FORMAT), "Europe/London")
```
Comment (andytaylorkainos) [inline core/builtins/custom_tags.py:68]: can you make "Europe/London" a constant please

File: core/builtins/custom_tags.py [hunk 2]
```diff
@@ -60,6 +62,29 @@ def str_date(value):
...
+    return date_str.strftime("%d %B %Y")
+
+
+@register.filter()
+def add_months(start_date, months):
+    start_date = datetime.datetime.strptime(start_date, "%Y-%m-%d")
```
Comment (andytaylorkainos) [inline core/builtins/custom_tags.py:74]: not going to use your DATE_FORMAT constant?

File: core/builtins/custom_tags.py [hunk 3]
```diff
@@ -60,6 +62,29 @@ def str_date(value):
...
+    year = start_date.year
+    month = start_date.month
+
+    for _ in range(months):
```
Comment (andytaylorkainos) [inline core/builtins/custom_tags.py:78]: really surprised that python's `datetime` doesnt have an easy way of doing this 🤯

File: core/builtins/custom_tags.py [hunk 4]
```diff
@@ -387,3 +412,8 @@ def abbreviate_string(string, length):
         return string
     else:
         return str(string)[:length] + "..."
+
+
+@register.filter()
+def get_usage_value(good):
+    return (good["usage"] / good["licenced_quantity"]) * float(good["licenced_value"])
```
Comment (andytaylorkainos) [inline core/builtins/custom_tags.py:419]: does this need to be rounded in any way?

File: licences/views.py [hunk 5]
```diff
@@ -45,3 +48,11 @@ def get(self, request, **kwargs):
             "row_limit": 3,
         }
         return render(request, "licences/licences.html", context)
+
+
+class Licence(TemplateView):
+    def get(self, request, pk):
+        licence, status_code = get_licence(request, pk)
+        if status_code != HTTPStatus.OK:
+            return Http404
```
Comment (andytaylorkainos) [inline licences/views.py:57]: would prefer if you were able to show a custom error message

Comment (BenAndrew-Kainos) [reply licences/views.py:57]: In this case this just means that the given ID doesn't exist, and so an error hasn't occurred but rather the page there looking for doesn't exist. hence the 404

Comment (BenAndrew-Kainos) [reply licences/views.py:57]: Can still change if you wish

File: core/builtins/custom_tags.py (hunk 4 above)
Comment (BenAndrew-Kainos) [reply core/builtins/custom_tags.py]: good point. done

File: core/builtins/custom_tags.py (hunk 3 above)
Comment (BenAndrew-Kainos) [reply core/builtins/custom_tags.py:78]: I was very surprised to find this out. Timedelta doesn't support months :(

File: core/builtins/custom_tags.py (hunk 2 above)
Comment (BenAndrew-Kainos) [reply core/builtins/custom_tags.py:74]: good point. done

File: core/builtins/custom_tags.py (hunk 1 above)
Comment (BenAndrew-Kainos) [reply core/builtins/custom_tags.py:68]: done

File: licences/views.py (hunk 5 above)
Comment (BenAndrew-Kainos) [reply licences/views.py:57]: Or I could have two different error outcomes

Comment (BenAndrew-Kainos) [reply licences/views.py:58]: Done

File: core/builtins/custom_tags.py [hunk 6]
```diff
@@ -387,3 +412,8 @@ def abbreviate_string(string, length):
         return string
     else:
         return str(string)[:length] + "..."
+
+
+@register.filter()
+def get_usage_value(good):
+    return round((good["usage"] / good["licenced_quantity"]) * float(good["licenced_value"]), 2)
```
Comment (efahy) [inline core/builtins/custom_tags.py:419]: Any need to protect from a divide by zero?

Comment (BenAndrew-Kainos) [reply core/builtins/custom_tags.py]: Removed this function

**Pull request merged**